docs/reference/configuration/configuration.md

```diff
@@ -1,48 +1,276 @@
-## Configuration overview
+## The configuration system
 
-The Arm Mbed configuration system customizes the compile time configuration of Mbed components: targets, libraries and applications.
+The Arm Mbed OS configuration system, a part of the Arm Mbed OS build tools, customizes compile time configuration parameters. Each library may define a number of configuration parameters in its `mbed_lib.json`. `mbed_app.json` may override the values of these configuration parameters. Configuration is defined using [JSON](http://www.json.org/). Some examples of configuration parameters:
 
-Mbed OS provides some default memory configurations for thread memory allocation. You may want to modify these configurations to better suit your project. You can read about the default configurations and how you can modify them below.
+- The sampling period for a data acquisition application.
+- The default stack size for a newly created OS thread.
+- The receive buffer size of a serial communication library.
+- The flash and RAM memory size of an Mbed target.
 
-#### Configuring the main thread stack
+The Arm Mbed OS configuration system gathers and interprets the configuration defined in the target in its [target configuration](/docs/development/tools/adding-and-configuring-targets.html), all `mbed_lib.json` files and the `mbed_app.json` file. The configuration system creates a single header file, `mbed_config.h`, that contains all of the defined configuration parameters converted into C preprocessor macros. `mbed compile` places `mbed_config.h` in the build directory, and `mbed export` places it in the application root. The Arm Mbed OS configuration system is run during `mbed compile` before invoking the compiler and during `mbed export` before creating project files.
```

Comment:
Query: Who or what runs the Arm Mbed OS configuration system during `mbed compile` before invoking the compiler? Please clarify for active voice.

Reply:
That's a very odd question. Also I think the answer will be irrelevant to the user. It's also circular: `mbed compile` runs the configuration system.

Reply:
You can word it similar to the prior sentence:

`mbed compile` run s the config sysetm ... and `mbed export` runs the config system...